platform/eureka/client.go

```diff
@@ -0,0 +1,135 @@
+// Copyright 2017 Istio Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package eureka
+
+import (
+	"encoding/json"
+	"fmt"
+	"io/ioutil"
+	"net/http"
+	"sort"
+	"time"
+)
+
+type application struct {
+	Name      string      `json:"name"`
+	Instances []*instance `json:"instance"`
+}
+
+type instance struct { // nolint: aligncheck
+	Hostname   string `json:"hostName"`
+	IPAddress  string `json:"ipAddr"`
+	Status     string `json:"status"`
+	Port       port   `json:"port"`
+	SecurePort port   `json:"securePort"`
+	// TODO: read dataCenterInfo for AZ support
+	Metadata metadata `json:"metadata,omitempty"`
+}
+
+type port struct {
```

Comment:
FWIW, pivotal doesn't support the older eureka servers. Maybe a documentation not on minimum supported eureka version?